### PR TITLE
Chore  add domain check informations

### PR DESCRIPTION
**Pull request by Kawacrepe**

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
 
-export function extractAllAuthorsFromLibrary(library: Scanner.Payload, flags: flagAuthor[] | null): authorsResponse
+export function extractAllAuthorsFromLibrary(library: Scanner.Payload, opts: options): Promise<authorsResponse[]>
+
+export interface options {
+  flags: flagAuthor[],
+  domainInformations: boolean,
+}
 
 export interface authorsResponse {
```
Comment (Rossb0b) [inline src/index.d.ts:11]: Should be renamed `authorResponse`

File: src/index.js [hunk 2]
```diff
@@ -53,7 +57,38 @@ export function extractAllAuthorsFromLibrary(library, flags = []) {
...
   }
 
-  return addFlagsInResponse(useLevenshtein(authors), flags);
+  const authorsWithFlags = addFlagsInResponse(useLevenshtein(authors), opts.flags);
+
+  if (opts.domainInformations === true) {
+    return addDomainInformations(authorsWithFlags);
+  }
+
+  return authorsWithFlags;
```
Comment (Rossb0b) [inline src/index.js:64]: I would use a ternary operator here but it's up to you

```
return opts.domainInformations ? addDomainInformations(authorsWithFlags) :
    authorsWithFlags;
```

File: src/index.d.ts [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
-  authors: Maintainer[] | [],
-  flags: flagAuthor[] | []
+  name?: string;
+  email?: string;
+  url?: string;
+  flagged: boolean,
+  packages: unknown[],
+  expirationDate: string,
+  mxRecords: unknown[],
```
Comment (Rossb0b) [inline src/index.d.ts:18]: Shouldn't be optional keys? And to me, it would be better if those keys were in a nested object as `domain`?

**Comment by Rossb0b** [comment]
I would have added some tests over rewriting it :
 - one without any options (flags & domainInformations)
 - one with flags
 - one with domainInformations
 
but it may be overkill

**Review by Rossb0b — COMMENTED** [review]
I would have added some tests over rewriting it :
- one without any options (flags & domainInformations)
- one with flags
- one with domainInformations

but it may be overkill

File: src/index.js (hunk 2 above)
Comment (Kawacrepe) [reply src/index.js:64]: I prefer to stay with the `if` version, since I don't know if I will add another opts later🤔

**Pull request merged**